purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -127,15 +127,30 @@ def _compute_qty_refunded(self):
         'move_ids.returned_move_ids.state',
     )
     def _compute_qty_returned(self):
+        """Made through read_group for not impacting in performance."""
+        StockMove = self.env['stock.move']
+        ProductUom = self.env['product.uom']
+        origin_moves = StockMove.search([
+            ('origin_returned_move_id.purchase_line_id', 'in', self.ids),
+        ])
+        groups = StockMove.read_group(
+            [('id', 'in', origin_moves.ids),
+             ('state', '=', 'done'),
+             ('location_id.usage', '!=', 'supplier')],
+            ['purchase_line_id', 'product_uom_qty', 'product_uom'],
+            ['purchase_line_id', 'product_uom'], lazy=False,
+        )
+        line_dict = {}
+        p = self._prefetch
+        for g in groups:
+            line_id = g['purchase_line_id'][0]
+            line_dict.setdefault(line_id, 0)
+            uom = ProductUom.browse(g['product_uom'][0]).with_prefetch(p)
```

Comment:
I think that if you want to prefetch those UoM, you have to browse all of them first in the same query.

i.e., add this at the top of the method:

```python
self.mapped("product_uom")
```

Then you can use here `ProductUom.browse(g['product_uom'][0]).with_prefetch(self._prefetch)`